app/scripts/controllers/network-order.ts

```diff
@@ -191,6 +189,13 @@ export class NetworkOrderController extends BaseController<
         // Append new networks to the end
         .concat(newNetworks);
     });
+
+    // The network controller can potentially update to a chain that is not selected in our enabled network map.
+    // This ensures that we fallback to a network that has been added to our network map.
+    const evmChainIds = Object.keys(
+      this.state.enabledNetworkMap[KnownCaipNamespace.Eip155],
+    );
+    this.#switchToEnabledNetworkIfNeeded(evmChainIds);
```

Comment:
I thought salim's fix for adding this `this.#switchToEnabledNetworkIfNeeded` would be sufficient, but it seems that some NetworkController state updates happen "after" we update the NetworkOrderController.

I'm open to opinions on this... part of me wants to handle this in the UI (through dispatches), but I don't have as much context on this.